Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp

```diff
@@ -435,6 +437,64 @@ int CLuaEngineDefs::EngineRestoreModel ( lua_State* luaVM )
     return 1;
 }
 
+int CLuaEngineDefs::EngineRequestModel ( lua_State* luaVM )
+{
+    SString strModelType;
+
+    CScriptArgReader argStream ( luaVM );
+    argStream.ReadString ( strModelType );
+
+    if ( !argStream.HasErrors () )
+    {
+        eClientModelType eModelType;
+        if ( stricmp( strModelType, "ped" ) == 0 )
+        {
+            eModelType = CCLIENTMODELPED;
+        }
+        else
+        {
+            lua_pushboolean ( luaVM, false );
+            return 1;
+        }
+
+        int iModelID = m_pManager->GetModelManager ()->GetFirstFreeModelID ();
+
+        CClientModel * pModel = new CClientModel( m_pManager, iModelID, eModelType );
+        pModel->Allocate ();
+        m_pManager->GetModelManager ()->Add ( pModel );
```

Comment:
You are already doing ```m_pModelManager->Add ( this );``` in the constructor of ```CClientModel```. 

Ideally we'd eliminate the use of new entirely and just use unique_ptrs for the managers in general to avoid memleaks entirely.